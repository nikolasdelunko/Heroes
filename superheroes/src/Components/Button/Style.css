.btn{
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  border: 2px solid #e74c3c;
  border-radius: 0.6em;
  color: #e74c3c;
  cursor: pointer;
  display: flex;
  align-self: center;
  font-size: 1rem;
  font-weight: 320;
  line-height: 1;
  margin: 20px;
  padding: 1.2em 2.8em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
}
.btn:hover, .btn:focus {
  color: #fff;
  outline: 0;
}

.first {
  -webkit-transition: box-shadow 300ms ease-in-out, color 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out, color 300ms ease-in-out;
}
.first:hover {
  box-shadow: 0 0 40px 40px #e74c3c inset;
}

.third {
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
}
.third:hover {
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}