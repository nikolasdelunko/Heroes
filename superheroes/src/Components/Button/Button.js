import "./Style.css";
import React from "react";

const Button = (props) => {
  const { classN, text, disabled, onClick, type } = props;
  return (
    <div className="btn-container">
      <button
        data-testid="button-element"
        className={classN}
        onClick={onClick}
        disabled={disabled}
        type={type}
      >
        {text}
      </button>
    </div>
  );
};

export default Button;
