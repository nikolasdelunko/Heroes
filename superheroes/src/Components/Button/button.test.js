import { render } from "@testing-library/react";
import Button from "./Button";

describe("Test Button", () => {
  test("Smoke test for Button", () => {
    render(<Button />);
  });

  test("Button shows datat if provided in props", () => {
    const { getByTestId, getByText } = render(
      <Button
        onClick={() => console.log("aa")}
        text={"textBtn"}
        disabled={false}
        type="submit"
        classN="control-btn form-btn"
      />
    );
		getByTestId("button-element")
		getByText("textBtn")
  });
});

describe("snapshot Testing", ()=> {
	test('Button snapshot', ()=> {
		const {container} = render(<Button />)
		expect(container.innerHTML).toMatchSnapshot()
	})
})
