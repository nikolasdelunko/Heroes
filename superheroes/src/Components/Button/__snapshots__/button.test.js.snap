// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Testing Button snapshot 1`] = `"<div class=\\"btn-container\\"><button data-testid=\\"button-element\\"></button></div>"`;
